cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -17,186 +17,108 @@ limitations under the License.
-
-	subCmdProperties := []struct {
-		use      string
-		short    string
-		long     string
-		examples string
-		cmdFunc  func(cfg *kubeadmapi.InitConfiguration, client clientset.Interface) error
-	}{
-		{
-			use:      "all",
-			short:    "Installs all addons to a Kubernetes cluster",
-			long:     allAddonsLongDesc,
-			examples: allAddonsExample,
-			cmdFunc:  EnsureAllAddons,
-		},
-		{
-			use:     "coredns",
-			short:   "Installs the CoreDNS addon to a Kubernetes cluster",
-			long:    corednsAddonsLongDesc,
-			cmdFunc: dnsaddon.EnsureDNSAddon,
-		},
-		{
-			use:     "kube-proxy",
-			short:   "Installs the kube-proxy addon to a Kubernetes cluster",
-			long:    kubeproxyAddonsLongDesc,
-			cmdFunc: proxyaddon.EnsureProxyAddon,
-		},
+// runEnsureCoreDNSAddon installs CoreDNS addon to a Kubernetes cluster
+func runEnsureCoreDNSAddon(c workflow.RunData) error {
+	cfg, client, err := getAddonData(c)
+	if err != nil {
+		return err
 	}
+	return dnsaddon.EnsureDNSAddon(cfg, client)
+}
 
-	for _, properties := range subCmdProperties {
-		// Creates the UX Command
-		cmd := &cobra.Command{
-			Use:     properties.use,
-			Short:   properties.short,
-			Long:    properties.long,
-			Example: properties.examples,
-			Run:     runAddonsCmdFunc(properties.cmdFunc, cfg, kubeConfigFile, &cfgPath, &featureGatesString),
-		}
-
-		// Add flags to the command
-		options.AddKubeConfigFlag(cmd.Flags(), &kubeConfigFile)
-		cmd.Flags().StringVar(&cfgPath, "config", cfgPath, "Path to a kubeadm config file. WARNING: Usage of a configuration file is experimental")
-		cmd.Flags().StringVar(&cfg.KubernetesVersion, "kubernetes-version", cfg.KubernetesVersion, `Choose a specific Kubernetes version for the control plane`)
-		cmd.Flags().StringVar(&cfg.ImageRepository, "image-repository", cfg.ImageRepository, `Choose a container registry to pull control plane images from`)
-
-		if properties.use == "all" || properties.use == "kube-proxy" {
-			cmd.Flags().StringVar(&cfg.APIEndpoint.AdvertiseAddress, "apiserver-advertise-address", cfg.APIEndpoint.AdvertiseAddress, `The IP address the API server is accessible on`)
-			cmd.Flags().Int32Var(&cfg.APIEndpoint.BindPort, "apiserver-bind-port", cfg.APIEndpoint.BindPort, `The port the API server is accessible on`)
-			cmd.Flags().StringVar(&cfg.Networking.PodSubnet, "pod-network-cidr", cfg.Networking.PodSubnet, `The range of IP addresses used for the Pod network`)
-		}
-
-		if properties.use == "all" || properties.use == "coredns" {
-			cmd.Flags().StringVar(&cfg.Networking.DNSDomain, "service-dns-domain", cfg.Networking.DNSDomain, `Alternative domain for services`)
-			cmd.Flags().StringVar(&cfg.Networking.ServiceSubnet, "service-cidr", cfg.Networking.ServiceSubnet, `The range of IP address used for service VIPs`)
-			cmd.Flags().StringVar(&featureGatesString, "feature-gates", featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
```

Comment:
I'm fine in removing this considering kubernetes/kubernetes: Pull Request 70024 will merge soon.
However, once this PR merges probably it will be necessary to add a new DNSType flag to allow users to choose between CoreDNS and kube-DNS.

@timothysc @neolit123  opinions?

Reply:
adding the CLI flag to chose DNS could be nice to have, then again we can avoid it.
in the discussions in this PR: https://github.com/kubernetes/website/pull/10294#discussion_r219645906
we discussed to drop GA feature gates that are doing toggling (like our CoreDNS gate) and to have a flag, instead.

then again, given with have it in the configuration the flag is not really needed, so my vote would be to only be able to pick DNS from the config.